File: app/src/main/java/com/github/pockethub/ui/user/HomePagerAdapter.java
/*
 * Copyright (c) 2015 PocketHub
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.github.pockethub.ui.user;

import android.content.res.Resources;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.view.ViewGroup;

import com.alorma.github.sdk.bean.dto.response.User;
import com.github.pockethub.R;
import com.github.pockethub.ui.FragmentPagerAdapter;
import com.github.pockethub.ui.repo.RepositoryListFragment;

import java.util.HashSet;
import java.util.Set;

/**
 * Pager adapter for a user's different views
 */
public class HomePagerAdapter extends FragmentPagerAdapter {

    private final User org;

    private boolean defaultUser;

    private final FragmentManager fragmentManager;

    private final Resources resources;

    private final Set<String> tags = new HashSet<>();

    /**
     * @param fragment
     * @param defaultUser
     */
    public HomePagerAdapter(final Fragment fragment,
        final boolean defaultUser, final User org) {
        super(fragment);

        this.org = org;
        fragmentManager = fragment.getChildFragmentManager();
        resources = fragment.getResources();
        this.defaultUser = defaultUser;
    }

    @Override
    public Fragment getItem(int position) {
        Fragment fragment = null;
        switch (position) {
            case 0:
                fragment = defaultUser ? new UserReceivedNewsFragment()
                    : new OrganizationNewsFragment();
                break;
            case 1:
                fragment = new RepositoryListFragment();
                break;
            case 2:
                fragment = defaultUser ? new MyFollowersFragment()
                    : new MembersFragment();
                break;
            case 3:
                fragment = new MyFollowingFragment();
                break;
        }

        if (fragment != null) {
            Bundle args = new Bundle();
            args.putParcelable("org", org);
            fragment.setArguments(args);
        }
        return fragment;
    }

    @Override
    public int getItemPosition(Object object) {
        return POSITION_NONE;
    }

    public Object instantiateItem(ViewGroup container, int position) {
        Object fragment = super.instantiateItem(container, position);
        if (fragment instanceof Fragment)
            tags.add(((Fragment) fragment).getTag());
        return fragment;
    }

    @Override
    public int getCount() {
        return defaultUser ? 4 : 3;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return resources.getString(R.string.tab_news);
            case 1:
                return resources.getString(R.string.tab_repositories);
            case 2:
                return resources.getString(defaultUser ? R.string.tab_followers_self
                    : R.string.tab_members);
            case 3:
                return resources.getString(R.string.tab_following_self);
            default:
                return null;
        }
    }
}

File: app/src/main/java/com/github/pockethub/android/ui/user/HomePagerAdapter.kt
/*
 * Copyright (c) 2015 PocketHub
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.github.pockethub.android.ui.user

import android.os.Bundle
import androidx.fragment.app.Fragment
import com.github.pockethub.android.R
import com.github.pockethub.android.ui.base.FragmentPagerAdapter
import com.github.pockethub.android.ui.repo.RepositoryListFragment
import com.meisolsson.githubsdk.model.User

/**
 * Pager adapter for a user's different views.
 */
class HomePagerAdapter(
        fragment: Fragment,
        private val defaultUser: Boolean,
        private val org: User
) : FragmentPagerAdapter(fragment) {

    private val resources = fragment.resources

    override fun getItem(position: Int): Fragment {
        val fragment = when (position) {
            0 -> if (defaultUser) UserReceivedNewsFragment() else OrganizationNewsFragment()
            1 -> RepositoryListFragment()
            2 -> if (defaultUser) MyFollowersFragment() else MembersFragment()
            3 -> MyFollowingFragment()
            else -> error("Unknown position: $position")
        }

        val args = Bundle()
        args.putParcelable("org", org)
        fragment.arguments = args

        return fragment
    }

    override fun getCount(): Int {
        return if (defaultUser) 4 else 3
    }

    override fun getPageTitle(position: Int): CharSequence {
        return when (position) {
            0 -> resources.getString(R.string.tab_news)
            1 -> resources.getString(R.string.tab_repositories)
            2 -> resources.getString(
                    if (defaultUser) R.string.tab_followers_self else R.string.tab_members
            )
            3 -> resources.getString(R.string.tab_following_self)
            else -> error("Unknown position: $position")
        }
    }
}
